src/data/string/basic.lean

```diff
@@ -70,10 +70,9 @@ by { cases s, refl }
 lemma to_list_nonempty {s : string} (h : s ≠ string.empty) :
   s.to_list = s.head :: (s.popn 1).to_list :=
 begin
-  rcases s with ⟨_ | ⟨hd, tl⟩⟩,
-  { simpa only using h },
-  { simp [to_list, popn, iterator.nextn, mk_iterator, head, iterator.next,
-          iterator.next_to_string, iterator.curr] }
+  induction s, cases s,
+  { exact absurd rfl h },
+  { exact rfl }
```

Comment:
The suggestion mechanism doesn't quite work here, but this can be golfed more:
```suggestion
lemma to_list_nonempty : ∀ {s : string}, s ≠ string.empty →
  s.to_list = s.head :: (s.popn 1).to_list
| ⟨s⟩ h := by cases s; [cases h rfl, refl]
```